lib/appengine/exec.rb

```diff
@@ -15,843 +15,11 @@
-
-      describe_build_strategy
-
-      config = build_config command, image, env_variables, cloud_sql_instances
-      file = ::Tempfile.new ["cloudbuild_", ".json"]
-      begin
-        ::JSON.dump config, file
-        file.flush
-        execute_command = [
-          "builds", "submit",
-          "--project", @project,
-          "--no-source",
-          "--config", file.path,
-          "--timeout", @timeout
-        ]
-        execute_command.concat ["--gcs-log-dir", @gcs_log_dir] unless @gcs_log_dir.nil?
-        Util::Gcloud.execute execute_command
-      ensure
-        file.close!
-      end
-    end
-
-    ##
-    # @private
-    # Workaround for https://github.com/GoogleCloudPlatform/appengine-ruby/issues/33
-    # Determines the image by looking it up in Cloud Build
-    #
-    def image_from_build app_info
-      create_time = ::DateTime.parse(app_info["createTime"]).to_time.utc
-      after_time = (create_time - 3600).strftime "%Y-%m-%dT%H:%M:%SZ"
-      before_time = (create_time + 3600).strftime "%Y-%m-%dT%H:%M:%SZ"
-      partial_uri = "gcr.io/#{@project}/appengine/#{@service}.#{@version}"
-      filter = "createTime>#{after_time} createTime<#{before_time} images[]:#{partial_uri}"
-      result = Util::Gcloud.execute \
-        [
-          "builds", "list",
-          "--project", @project,
-          "--filter", filter,
-          "--format", "json"
-        ],
-        capture: true, assert: false
-      result.strip!
-      raise NoSuchVersion.new(@service, @version) if result.empty?
-      build_info = ::JSON.parse(result).first
-      build_info["images"].first
-    end
-
-    def describe_build_strategy
-      puts "\nUsing the `cloud_build` strategy for appengine:exec"
-      puts "(i.e. running your app image in Cloud Build)"
-      puts "PROJECT: #{@project}"
-      puts "SERVICE: #{@service}"
-      puts "VERSION: #{@version}"
-      puts "TIMEOUT: #{@timeout}"
-      puts ""
-    end
-
-    ##
-    # @private
-    # Builds a cloudbuild config as a data structure.
-    #
-    # @param command [Array<String>] The command in array form.
-    # @param image [String] The fully qualified image path.
-    # @param env_variables [Hash<String,String>] Environment variables.
-    # @param cloud_sql_instances [String,Array<String>] Names of cloud sql
-    #     instances to connect to.
-    #
-    def build_config command, image, env_variables, cloud_sql_instances
-      args = ["-i", image]
-      env_variables.each do |k, v|
-        v = v.gsub "$", "$$"
-        args << "-e" << "#{k}=#{v}"
-      end
-      unless cloud_sql_instances.empty?
-        cloud_sql_instances = Array(cloud_sql_instances)
-        cloud_sql_instances.each do |sql|
-          args << "-s" << sql
-        end
-      end
-      args << "--"
-      args += command
-
-      {
-        "steps" => [
-          "name" => @wrapper_image,
-          "args" => args
-        ]
-      }
-    end
-  end
-end
+AppEngine::Exec = Google::Serverless::Exec
```

Comment:
I recommend

```
module AppEngine
  Exec = Google::Serverless::Exec
end
```

Just in case the AppEngine module hasn't yet been defined at the time this file is evaluated.